task4/main.c

```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifndef MAX_BUF
+#define MAX_BUF 999 // USED ​​IN FUNCTION fgets()
+#endif
+
+void freeing_memory(char **text, int size) {
+  for (int i = 0; i < size; i++) {
+    free(text[i]);
+  }
+  free(text);
+}
+
+int main() {
+  int len, lines_number = 0;
+  char **text;
+  text = (char **)malloc(1 * sizeof(char *));
+  if (text == NULL) {
+    perror("Error: ");
```

Comment:
По-моему, сообщение будет выглядеть "Error: : some error". Может, лучше написать
```suggestion
    perror("Error allocating memory");
```